presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/customviews/DashboardFragment.kt

```diff
@@ -18,4 +22,26 @@ class DashboardFragment : Fragment() {
         binding = FragmentDashboardBinding.inflate(inflater, container, false)
         return binding.root
     }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initObservers()
+    }
+
+    private fun initObservers() {
+        viewModel.isExistUser.observe(viewLifecycleOwner) { isUserLogged ->
+            if(isUserLogged) {
+                binding.backButton.visibility = View.GONE
+            } else {
+                binding.backButton.customButtonListener {
+                    val intent = Intent()
+                    intent.setClassName(
+                        requireContext(),
+                        "com.safetyheads.akademiaandroida.presentation.ui.MainActivity"
+                    )
+                    startActivity(intent)
```

Comment:
Please add "finish()" these activity. I think that "finish()" in MainActivity (when open DashboardActivity) will fix the error when we come from Sign in and return next